### ISSUE TITLE
Resnet 50 fails in ToTtPass on first batch normalization

### ISSUE THREAD

**Issue by maximiliankir** [tenstorrent/pytorch2.0_ttnn#727]
I try to run the compilation for a ResNet50 model:

```Python
import torch
import ttnn
import torchvision.models as models
import torch_ttnn

# Load ResNet model
weights = models.ResNet50_Weights.DEFAULT
model = models.resnet50(weights=weights)
model = model.to(dtype=torch.bfloat16)
model.eval()

# Compile the model
device = ttnn.open_device(device_id=0)
option = torch_ttnn.TorchTtnnOption(device=device, verbose=True)
ttnn_model = torch.compile(model, backend=torch_ttnn.backend, options=option)

# Run inference
input = torch.rand(1, 3, 224, 224, dtype=torch.bfloat16)

output = ttnn_model(input)
```

I get an exception in the ToTtPass:
```
torch._dynamo.exc.BackendCompilerFailed: backend='ttnn_backend' raised:
Exception: An error occurred when running the 'ToTtPass' pass after the following passes: ['ConstantFoldingPass']
```

I tried to investigate this. The exception comes from the assertion in to_tt_pass.py:447

```Python
assert is_getitem_0_only_user(node), "non-training batch_norm should only have first return value used"
```
The batch normalization actually has 3 users called "getitem", "getitem_1" and "getitem_2".

Probably I am doing something wrong here, because Resnet50 seems to work in the tests in the CI pipeline. Can someone please help me with this?

**Comment by kevinwuTT** [comment]
Can you try running your code with these changes?

This should fix the error you're seeing:

```diff
- output = ttnn_model(input)
+ with torch.no_grad():
+     output = ttnn_model(input)
```

This should fix OOM error: `Out of Memory: Not enough space to allocate 3584 B L1_SMALL buffer across 56 banks, where each bank needs to store 64B`

```diff
- device = ttnn.open_device(device_id=0)
+ device = ttnn.open_device(device_id=0, l1_small_size=16384)
```

**Comment by maximiliankir** [comment]
Thank you this resolved the exception. But I am still not able to run the Resnet50 model. 
The compilation is successful, but the execution hangs. It seems to hang on the ttnn.permute operation.
Which is similar to the issue, that I already raised in https://github.com/tenstorrent/tt-metal/issues/17137

**Comment by kevinwuTT** [comment]
Yeah, I was able to run your code with the changes with no errors. As of this writing, I'm running on main (https://github.com/tenstorrent/pytorch2.0_ttnn/commit/cdbed3f6d00a6c22bdec4cd2e54f2912c79337cf) with ttnn [0.55.0rc20](https://github.com/tenstorrent/pytorch2.0_ttnn/blob/cdbed3f6d00a6c22bdec4cd2e54f2912c79337cf/requirements.txt#L9), and dependencies from [requirements-dev.txt](https://github.com/tenstorrent/pytorch2.0_ttnn/blob/main/requirements-dev.txt). My system is Ubuntu 20.04.6 using a TT n150 on firmware bundle version 80.13.0.0.

**Comment by maximiliankir** [comment]
I'm not sure what the problem was for me, but it is running now. I updated my kernel mode driver, the firmware and reinstalled pytorch_ttnn with all its dependencies. And it is running now. Thanks for your help.
